src/frontend/src/pages/WorkPackageForm/WorkPackageForm.tsx

```diff
@@ -0,0 +1,76 @@
+import { UseMutationResult } from 'react-query';
+import { WbsNumber, WorkPackage, isGuest, isProject, wbsPipe } from 'shared';
+import WorkPackageFormView, { WorkPackageFormViewPayload } from './WorkPackageFormView';
+import { bulletsToObject } from '../../utils/form';
+import { useAllWorkPackages } from '../../hooks/work-packages.hooks';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import ErrorPage from '../ErrorPage';
+import { useAllUsers } from '../../hooks/users.hooks';
+import { useSingleProject } from '../../hooks/projects.hooks';
+
+interface WorkPackageFormProps {
+  wbsNum: WbsNumber;
+  operation: (wbsNum: WbsNumber) => UseMutationResult;
+  exitActiveMode: () => void;
+  crId?: string;
+}
+
+const WorkPackageForm: React.FC<WorkPackageFormProps> = ({ wbsNum, operation, exitActiveMode, crId }) => {
+  const { data: users, isLoading: usersIsLoading, isError: usersIsError, error: usersError } = useAllUsers();
+  const {
+    data: project,
+    isLoading: projectIsLoading,
+    isError: projectIsError,
+    error: projectError
+  } = useSingleProject({ ...wbsNum, workPackageNumber: 0 });
+  const { data: workPackages, isLoading: wpIsLoading, isError: wpIsError, error: wpError } = useAllWorkPackages();
+  const { mutateAsync } = operation(wbsNum);
+
+  if (wpIsLoading || !workPackages || usersIsLoading || !users || projectIsLoading || !project) return <LoadingIndicator />;
+  if (usersIsError) return <ErrorPage message={usersError.message} />;
+  if (projectIsError) return <ErrorPage message={projectError.message} />;
+  if (wpIsError) return <ErrorPage message={wpError.message} />;
+
+  const workPackage = workPackages.find(
+    (wp) =>
+      wp.wbsNum.carNumber === wbsNum.carNumber &&
+      wp.wbsNum.projectNumber === wbsNum.projectNumber &&
+      wp.wbsNum.workPackageNumber === wbsNum.workPackageNumber
+  );
+
+  const defaultValues: WorkPackageFormViewPayload | undefined = isProject(wbsNum)
```

Comment:
you should pass in a prop that determines whether the form is a create or edit form to determine stuff like this instead of doing isProject(wbsNum). This way does work, but it relies on an invariant that we just happen to be enforcing rn, so better to make it more explicit